### PR TITLE
move to Kubernetes v1.18.2 API

### LINKED ISSUE #317: Adapt to K8s 1.18 API changes
Ensure K8s 1.18 API changes are taken into account in Intel device plugins.

### PR DESCRIPTION
**Pull request by rojkov**
closes #317

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -5,47 +5,47 @@ go 1.13
 require (
 	github.com/fsnotify/fsnotify v1.4.7
 	github.com/go-ini/ini v1.46.0
+	github.com/golang/glog v0.0.0-20160126235308-23def4e6c14b
```
Comment (mythi) [inline go.mod:8]: do you know why this gets pulled in?

Comment (rojkov) [reply go.mod:8]: Not yet. `go mod tidy` doesn't help, it keeps creeping in.

**Comment by mythi** [comment]
perhaps we should take #364 in this PR too?

**Comment by mythi** [comment]
can we document the steps to get 453ad1c easily done in `DEVEL.md`?

**Comment by rojkov** [comment]
> can we document the steps to get [453ad1c](https://github.com/intel/intel-device-plugins-for-kubernetes/commit/453ad1cad76972bfddd22d418c3bff658da3250c) easily done in `DEVEL.md`?

Sure, I'll update DEVEL.md FWIW

Usually I follow this post https://www.openshift.com/blog/kubernetes-deep-dive-code-generation-customresources. Though it's assumed `dep` is used instead of `go mod`. The latter makes the steps look like black magic:
1. bump up the deps,
2. copy the repo to $GOPATH/srd/github.com/intel/...,
3. put the boilerplate to the place k8s's codegen script expects it (no way to specify it as an option),
4. run the script
5. copy the generate code back to the original repository
6. fix compilation issues.

Probably there's a better way.

**Comment by rojkov** [comment]
> perhaps we should take #364 in this PR too?

I tried that, but it turned out v1beta1 and v1 have substantial differences. Here's an example for the two versions
https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#create-a-customresourcedefinition
E2E tests fail if to change the version only. I'd rather fix the issue in a followup PR.

File: go.mod (hunk 1 above)
Comment (mythi) [reply go.mod]: looks like this is fixed

**Comment by rojkov** [comment]
Oops, I didn't notice github automatically requested your reviews. Now it's ready for review.

File: go.mod (hunk 1 above)
Comment (rojkov) [reply go.mod]: Yes, `go mod why` is a great tool to solve mysteries. Turned out it was a temp folder with my hacks that brought the dep.

**Comment by bart0sh** [comment]
@mythi can you merge?

**Comment by mythi** [comment]
the CI returns an error. perhaps retrigger the build to make sure it's OK?

**Comment by rojkov** [comment]
> the CI returns an error. perhaps retrigger the build to make sure it's OK?

It seems I don't have permissions to restart jenkins pipelines.

**Comment by mythi** [comment]
> It seems I don't have permissions to restart jenkins pipelines.

I usually just force push with a new commit id.

**Comment by bart0sh** [comment]
@mythi now it seems to be ready for merging.

**Pull request merged**